src/libraries/System.Net.Http/tests/FunctionalTests/MultipartContentTest.cs

```diff
@@ -387,6 +388,128 @@ public void ReadAsStream_CreateContentReadStreamThrows()
             Assert.Throws<NotImplementedException>(() => mc.ReadAsStream());
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task ReadAsStream_CustomEncodingSelector_SelectorIsCalledWithCustomState(bool async)
```

Comment:
NIT: I left the naming as `ReadAsStreamAsync_...` when I added the sync code path since it still is an `async` test method.

Reply:
Changed